Code/Framework/AzFramework/AzFramework/Terrain/TerrainDataRequestBus.h

```diff
@@ -130,10 +130,16 @@ namespace AzFramework
                 SurfaceData::SurfacePoint& outSurfacePoint,
                 Sampler sampleFilter = Sampler::DEFAULT,
                 bool* terrainExistsPtr = nullptr) const = 0;
+
+        private:
+            // Functions without the optional bool* parameter that can be used from Python tests.
+            virtual float GetHeightVal(AZ::Vector3 position, Sampler sampler = Sampler::BILINEAR) const = 0;
+            virtual float GetHeightValFromVector2(AZ::Vector2 position, Sampler sampler = Sampler::BILINEAR) const = 0;
+            virtual float GetHeightValFromFloats(float x, float y, Sampler sampler = Sampler::BILINEAR) const = 0;
```

Comment:
Rather than adding them as virtual here and putting the implementation in Terrain System, how about making them non-virtual private and putting the implementation with this class instead?  That way we don't have to add these to mocks or anything else that inherits from this class.